core/lib/src/unix.rs

```diff
@@ -0,0 +1,67 @@
+use crate::error::{LaunchError, LaunchErrorKind};
+use std::path::Path;
+use std::fs::{self, File};
+
+#[doc(hidden)]
+pub fn lock_socket<P: AsRef<Path>>(path: P) -> Result<File, LaunchError> {
+    use fs2::FileExt;
+
+    let path = path.as_ref();
+    let lock_path = path.with_extension("lock");
+
+    let lock_file = fs::OpenOptions::new()
+        .read(true).write(true).create(true)
+        .open(&lock_path)?;
+
+    lock_file
+        .try_lock_exclusive()
+        .map_err(|e| LaunchError::new(LaunchErrorKind::FailedSocketLock(lock_path, e)))?;
+
+    // On Unix, the file remains until the last handle to it is closed. We
+    // return the handle to the user. When it is dropped, the lock will free.
+    if path.exists() {
+        fs::remove_file(path)?;
```

Comment:
I don't understand; what takes care of removing the lock file? What is keeping a handle to the file at `path` alive (we return a handle to `lock_file`, i.e, `lock_path`, not `path`)? Is the file guaranteed to be deleted even if the user force-terminates the server with, say, `ctrl-c`?

Reply:
1. The lock file isn't removed,
2. `path` is only used for generating `lock_path`,
3. Removing `path` is for ensuing that Unix socket can listen on that path(it will throw an error if the path exists beforehand).

Reply:
_Can_ we remove the lock file? it seems unfortunate to leave files around.

Reply:
I don't think we can do that now. Removing the lock file automatically would require some shutdown cleanup process (#180) which also isn't guaranteed if process got killed by SIGKILL